pkg/discovery/network/generic_test.go

```diff
@@ -157,6 +167,115 @@ var _ = Describe("discoverGenericNetwork", func() {
 			Expect(clusterNet.NetworkPlugin).To(BeIdenticalTo("generic"))
 		})
 	})
+
+	When("There is no pod cidr information on node", func() {
+		It("Should return nil cluster network", func() {
+			clusterNet := testDiscoverGenericWith(
+				fakeNode("node1", ""),
+				fakeNode("node2", ""),
+			)
+			Expect(clusterNet).To(BeNil())
+		})
+	})
+
+	When("There is pod cidr information on any of nodes", func() {
+		var clusterNet *ClusterNetwork
+
+		BeforeEach(func() {
+			clusterNet = testDiscoverGenericWith(
+				fakeNode("node1", ""),
+				fakeNode("node2", testPodCIDR),
+			)
+		})
+
+		It("Should return the ClusterNetwork structure with PodCIDR", func() {
+			Expect(clusterNet.PodCIDRs).To(Equal([]string{testPodCIDR}))
+		})
+
+		It("Should identify the networkplugin as generic", func() {
+			Expect(clusterNet.NetworkPlugin).To(BeIdenticalTo("generic"))
+		})
+
+		It("Should return the ClusterNetwork structure with empty service CIDR", func() {
+			Expect(clusterNet.ServiceCIDRs).To(BeEmpty())
+		})
+	})
+
+	When("There is pod cidr information on any of nodes and there is kubeapi pod", func() {
+		var clusterNet *ClusterNetwork
+
+		BeforeEach(func() {
+			clusterNet = testDiscoverGenericWith(
+				fakeNode("node1", testPodCIDR),
+				fakePod("kube-apiserver", []string{"kube-apiserver", "--service-cluster-ip-range=" + testServiceCIDR}, []v1.EnvVar{}),
+			)
+		})
+
+		It("Should return ClusterNetwork with all CIDRs", func() {
+			Expect(clusterNet.ServiceCIDRs).To(Equal([]string{testServiceCIDR}))
+			Expect(clusterNet.PodCIDRs).To(Equal([]string{testPodCIDR}))
+		})
+
+		It("Should identify the networkplugin as generic", func() {
+			Expect(clusterNet.NetworkPlugin).To(BeIdenticalTo("generic"))
+		})
+	})
+
+	When("No resource and creating invalid svc doesn't return error", func() {
+		It("Should return nil cluster network", func() {
+			clusterNet := testDiscoverGenericEmulatedErrWith(nil)
+			Expect(clusterNet).To(BeNil())
+		})
+	})
+
+	When("No resource and creating invalid svc returns unexpected error", func() {
+		It("Should return nil cluster network", func() {
+			clusterNet := testDiscoverGenericEmulatedErrWith(errUnexpected)
+			Expect(clusterNet).To(BeNil())
+		})
+	})
+
+	When("No resource and creating invalid svc returns expected error", func() {
+		var clusterNet *ClusterNetwork
+
+		BeforeEach(func() {
+			clusterNet = testDiscoverGenericEmulatedErrWith(errExpected)
+		})
+
+		It("Should return the ClusterNetwork structure with PodCIDR", func() {
+			Expect(clusterNet.PodCIDRs).To(BeEmpty())
+		})
+
+		It("Should identify the networkplugin as generic", func() {
+			Expect(clusterNet.NetworkPlugin).To(BeIdenticalTo("generic"))
+		})
+
+		It("Should return the ClusterNetwork structure with empty service CIDR", func() {
+			Expect(clusterNet.ServiceCIDRs).To(Equal([]string{testServiceCIDR}))
+		})
+
+	})
+
+	When("Node with cidr information and creating invalid svc returns expected error", func() {
```

Comment:
This is essentially the same as the case on L204 when both exist in some manner.